okio/jvm/src/main/java/okio/RealBufferedSource.kt

```diff
@@ -34,24 +34,25 @@ internal class RealBufferedSource(
   override fun buffer() = bufferField
 
   override fun read(sink: Buffer, byteCount: Long): Long {
-    require(byteCount >= 0) { "byteCount < 0: $byteCount" }
-    check(!closed) { "closed" }
-
-    if (buffer.size == 0L) {
-      val read = source.read(buffer, Segment.SIZE.toLong())
-      if (read == -1L) return -1L
-    }
-
-    val toRead = minOf(byteCount, buffer.size)
-    return buffer.read(sink, toRead)
+    return readMaybeAsync(sink, byteCount, Source::read)
   }
 
   override suspend fun readAsync(sink: Buffer, byteCount: Long): Long {
+    // Can't use a suspend function reference here apparently.
```

Comment:
Hmm, is this a type inference problem, or a limitation of coroutines API?

Reply:
I think it's the former. The error message is something like:
```
Type mismatch: inferred type is KSuspendFunction0<Buffer, Long, Long> but (Buffer, Long) -> Long was expected
```

Reply:
Might be worth opening an issue on Kotlin's YouTrack.

Reply:
I just filed this issue: https://youtrack.jetbrains.net/issue/KT-28014